bin/crew

```diff
@@ -496,89 +469,61 @@ def cache_build
   else
-    @ruby_default_constants = %w[
-      ARGF
-      ARGV
-      CROSS_COMPILING
-      DOC
-      ENV
-      GC
-      IO
-      JSON
-      OpenSSL
-      Q
-      R
-      RUBY_COPYRIGHT
-      RUBY_DESCRIPTION
-      RUBY_ENGINE
-      RUBY_ENGINE_VERSION
-      RUBYGEMS_ACTIVATION_MONITOR
-      RUBY_PATCHLEVEL
-      RUBY_PLATFORM
-      RUBY_RELEASE_DATE
-      RUBY_REVISION
-      RUBY_VERSION
-      RubyVM
-      S
-      STDERR
-      STDIN
-      STDOUT
-      StringIO
-      TOPLEVEL_BINDING
-      URI
-    ]
-    # Get all constants
-    @constants = Module.constants.grep(/[[:upper:]]$/)
-    # Reject all constants which match the default list
-    @constants = @constants.map(&:to_s).reject { |e| @ruby_default_constants.any? { |f| /\A#{e}\z/ =~ f } }
+    # Get all constants defined by const.rb
+    constants = Object.constants.select do |const|
+      Object.const_source_location(const)[0] == File.join(CREW_LIB_PATH, 'lib/const.rb')
+    end.sort
+
     # Print a sorted list of the remaining constants used by crew.
-    @constants.sort.each do |var|
-      value = eval(var.to_s)
+    constants.each do |var|
+      value = Object.const_get(var.to_sym)
       puts "#{var}=#{value}"
     end
   end
 end
 
 def files(pkgName)
-  filelist = "#{CREW_META_PATH}#{pkgName}.filelist"
-  if File.file? filelist
-    system "sort #{filelist}"
-    lines = File.readlines(filelist).size
+  filelist_path = File.join(CREW_META_PATH, "#{pkgName}.filelist")
+  if File.file?(filelist_path)
+    filelist = File.readlines(filelist_path, chomp: true).sort
+    lines = filelist.size
     size = 0
-    File.readlines(filelist).each do |filename|
-      size += File.size(filename.chomp) if File.file? filename.chomp
+
+    filelist.each do |filename|
+      # ignore symlinks to prevent duplicating calculation
+      size += File.size(filename) if File.file?(filename) && !File.symlink?(filename)
     end
-    humansize = human_size(size)
-    puts "Total found: #{lines}".lightgreen
-    puts "Disk usage: #{humansize}".lightgreen
+
+    puts filelist, <<~EOT.lightgreen
+
+      Total found: #{lines}
+      Disk usage: #{human_size(size)}
+    EOT
   else
     puts "Package #{pkgName} is not installed. :(".lightred
   end
 end
 
 def whatprovides(regexPat)
-  # Use grep version command to ascertain whether we have a working grep.
-  abort 'Grep is not working. Please install it with \'crew install grep\''.lightred unless system('grep -V > /dev/null 2>&1')
-  fileArray = []
-  @grepresults = `grep "#{regexPat}" #{CREW_META_PATH}*.filelist`.chomp.gsub('.filelist', '').gsub(':', ': ').gsub(CREW_META_PATH, '').split(/$/).map(&:strip)
-  @grepresults.each { |fileLine| fileArray.push(fileLine) }
-  unless fileArray.empty?
-    fileArray.sort.each do |item|
-      puts item
-    end
-    puts "\nTotal found: #{fileArray.length}".lightgreen
-  end
+  matchedList = `grep "#{regexPat}" #{CREW_META_PATH}*.filelist`.lines(chomp: true).flat_map do |result|
```

Comment:
Can we assume grep is always available?

Reply:
Yes, `/bin/grep` should available out of box